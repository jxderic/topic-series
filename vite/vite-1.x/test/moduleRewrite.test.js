const {
  transformCjsImport
} = require('../dist/node/server/serverPluginModuleRewrite')

describe('rewrite commonjs imports', () => {
  const cases = [
    `import React from 'react'`,
    `import * as React from 'react'`,
    `import React, { useState } from 'react'`,
    `import { useState } from 'react'`,
    `import { useState, useEffect } from 'react'`,
    `import { useState as something } from 'react'`,
    `import { useState as something, useEffect } from 'react'`,
    `import { useState as something, useEffect as alias } from 'react'`,
    `import { default as Default } from 'react'`,
    `import { default as Default, useEffect } from 'react'`
  ]
  for (let [testCase, i] of cases.map((x, i) => [x, i])) {
    test(`${i} "${testCase}"`, () => {
      const res = transformCjsImport(testCase, 'react', 'react', 0)
      expect(res).toMatchSnapshot()
    })
  }
})
