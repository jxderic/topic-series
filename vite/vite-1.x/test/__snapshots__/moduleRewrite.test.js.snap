// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rewrite commonjs imports 0 "import React from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const React = $viteCjsImport0_react;"
`;

exports[`rewrite commonjs imports 1 "import * as React from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const React = $viteCjsImport0_react;"
`;

exports[`rewrite commonjs imports 2 "import React, { useState } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const React = $viteCjsImport0_react;
const useState = $viteCjsImport0_react[\\"useState\\"];"
`;

exports[`rewrite commonjs imports 3 "import { useState } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const useState = $viteCjsImport0_react[\\"useState\\"];"
`;

exports[`rewrite commonjs imports 4 "import { useState, useEffect } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const useState = $viteCjsImport0_react[\\"useState\\"];
const useEffect = $viteCjsImport0_react[\\"useEffect\\"];"
`;

exports[`rewrite commonjs imports 5 "import { useState as something } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const something = $viteCjsImport0_react[\\"useState\\"];"
`;

exports[`rewrite commonjs imports 6 "import { useState as something, useEffect } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const something = $viteCjsImport0_react[\\"useState\\"];
const useEffect = $viteCjsImport0_react[\\"useEffect\\"];"
`;

exports[`rewrite commonjs imports 7 "import { useState as something, useEffect as alias } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const something = $viteCjsImport0_react[\\"useState\\"];
const alias = $viteCjsImport0_react[\\"useEffect\\"];"
`;

exports[`rewrite commonjs imports 8 "import { default as Default } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const Default = $viteCjsImport0_react;"
`;

exports[`rewrite commonjs imports 9 "import { default as Default, useEffect } from 'react'" 1`] = `
"import $viteCjsImport0_react from \\"react\\";
const Default = $viteCjsImport0_react;
const useEffect = $viteCjsImport0_react[\\"useEffect\\"];"
`;
